### PR TITLE
Make Mapping covariant.

### LINKED ISSUE #510: Make Mapping covariant
I think that `Mapping[KT, VT]` should be covariant in both `KT` and `VT`. The argument for this is that it doesn't have any mutation operations. Mypy might complain (or may eventually start to complain) about having covariant type variables in argument types but in this case it's okay so we can probably just `# type: ignore` them.

(It wouldn't be safe since somebody can write a `get` method that actually mutates the mapping, but consenting adults...)

### PR DESCRIPTION
**Pull request by gvanrossum**
Fixes #510.

This requires a `# type: ignore` on `__getitem__` and `get` because
mypy complains about covariant parameters.

FWIW typing.py needs to be changed too.  (It was covariant in the
value but invariant in the key -- typeshed was invariant in both.)

See also https://github.com/python/typing/pull/273.

### CODE REVIEW

File: stdlib/3/typing.pyi [hunk 1]
```diff
@@ -226,16 +226,18 @@ class ValuesView(MappingView, Iterable[_VT_co], Generic[_VT_co]):
 
 # TODO: ContextManager (only if contextlib.AbstractContextManager exists)
 
-class Mapping(Iterable[_KT], Container[_KT], Sized, Generic[_KT, _VT]):
+class Mapping(Iterable[_KT_co], Container[_KT_co], Sized, Generic[_KT_co, _VT_co]):
     # TODO: Value type should be covariant, but currently we can't give a good signature for
```
Comment (JukkaL) [inline stdlib/3/typing.pyi]: The comment comment is out of date.

File: stdlib/2.7/typing.pyi [hunk 2]
```diff
@@ -176,17 +176,19 @@ class ValuesView(MappingView, Iterable[_VT_co], Generic[_VT_co]):
     def __contains__(self, o: object) -> bool: ...
     def __iter__(self) -> Iterator[_VT_co]: ...
 
-class Mapping(Sized, Iterable[_KT], Container[_KT], Generic[_KT, _VT]):
-    @abstractmethod
-    def __getitem__(self, k: _KT) -> _VT: ...
+class Mapping(Sized, Iterable[_KT_co], Container[_KT_co], Generic[_KT_co, _VT_co]):
+    @abstractmethod  # type: ignore
+    def __getitem__(self, k: _KT_co) -> _VT_co:  # type: ignore
```
Comment (JukkaL) [inline stdlib/2.7/typing.pyi]: Add comment about the `# type: ignore` comments (also to the Python 3 stub).

File: stdlib/3/typing.pyi (hunk 1 above)
Comment (gvanrossum) [inline stdlib/3/typing.pyi]: Thanks, pushed a fix deleting the comment.

**Comment by gvanrossum** [comment]
OK, as a compromise I am making it covariant in the value only. (It was already that way in typing.py.) There's still a `# type: ignore` on the definition of get(), mypy doesn't believe the hack @JukkaL suggested yet.

**Pull request merged**